src/web3/SmartContractsClient.ts

```diff
@@ -271,7 +325,20 @@ export class SmartContractsClient
     }
   }
 
-  /** Read-only smart contracts */
+  /**
+   * Execute a read-only smart contract.
+   *
+   * @param contractData - The data required for the read-only smart contract.
+   *
+   * @returns A promise which resolves to an object containing the result
+   * of the operation.
+   *
+   * @throws
+   * - If the contract binary data is missing.
+   * - If the contract contract address is missing.
+   * - If the result is empty.
+   * - If the result contains an error.
+   */
   public async executeReadOnlySmartContract(
     contractData: IContractData,
   ): Promise<IExecuteReadOnlyResponse> {
```

Comment:
@AurelienFT Could you explain how execute_read_only_bytecode works ?

Reply:
@Leo-Besancon could you help us on that ?

Reply:
@Ben-Rey I'll probably have some time for you tomorrow, but I'm not sure how much help I can bring about this function ;)

Reply:
@Ben-Rey Apologies, I missed the notification. A RO call or execution in massa has the same behaviour as a normal call or execution with the exception that every change made on the ledger by the given call / execution will get reverted after completion. Also another important thing to note is that a RO request will not create an operation (like CallSC or ExecuteSC would), it is considered as low priority execution, since it emits no operation it has no trace in blocks, and only one RO request can be executed per slot. Let me know if that answers your question.